dot/network/authoritydiscovery/service.go

```diff
@@ -0,0 +1,303 @@
+type ServiceToWorkerMsg interface {
+	serviceToWorkerMsg()
+}
+
+// GetAddressesByAuthorityID is a message to get addresses for an authority
+type GetAddressesByAuthorityID struct {
+	AuthorityID AuthorityID
+	Response    chan<- *AddressesResponse
+}
+
+func (GetAddressesByAuthorityID) serviceToWorkerMsg() {}
+
+// GetAuthorityIDsByPeerID is a message to get authority IDs for a peer
+type GetAuthorityIDsByPeerID struct {
+	PeerID   peer.ID
+	Response chan<- *AuthorityIDsResponse
+}
+
+func (GetAuthorityIDsByPeerID) serviceToWorkerMsg() {}
+
+// AddressesResponse represents the response for address queries
+type AddressesResponse struct {
+	Addresses []ma.Multiaddr
+	Err       error
+}
+
+// AuthorityIDsResponse represents the response for authority ID queries
+type AuthorityIDsResponse struct {
+	AuthorityIDs []AuthorityID
+	Err          error
+}
+
+// Service provides the public API for authority discovery
+type Service struct {
+	toWorker chan<- ServiceToWorkerMsg
+	closed   bool
+	mu       sync.RWMutex
+}
+
+// NewService creates a new Service instance
+func NewService(toWorker chan<- ServiceToWorkerMsg) *Service {
+	return &Service{
+		toWorker: toWorker,
+		closed:   false,
+	}
+}
+
+// GetAddressesByAuthorityID retrieves addresses associated with an authority ID
+func (s *Service) GetAddressesByAuthorityID(ctx context.Context, authority AuthorityID) ([]ma.Multiaddr, error) {
+	s.mu.RLock()
+	if s.closed {
+		s.mu.RUnlock()
+		return nil, ErrServiceClosed
+	}
+	s.mu.RUnlock()
+
+	response := make(chan *AddressesResponse, 1)
+	msg := GetAddressesByAuthorityID{
+		AuthorityID: authority,
+		Response:    response,
+	}
+
+	select {
+	case s.toWorker <- msg:
+		select {
+		case resp := <-response:
+			if resp.Err != nil {
+				return nil, resp.Err
+			}
+			return resp.Addresses, nil
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		}
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	}
+}
+
+// GetAuthorityIDsByPeerID retrieves authority IDs associated with a peer ID
+func (s *Service) GetAuthorityIDsByPeerID(ctx context.Context, peerID peer.ID) ([]AuthorityID, error) {
+	s.mu.RLock()
+	if s.closed {
+		s.mu.RUnlock()
+		return nil, ErrServiceClosed
+	}
+	s.mu.RUnlock()
+
+	response := make(chan *AuthorityIDsResponse, 1)
+	msg := GetAuthorityIDsByPeerID{
+		PeerID:   peerID,
+		Response: response,
+	}
+
+	select {
+	case s.toWorker <- msg:
+		select {
+		case resp := <-response:
+			if resp.Err != nil {
+				return nil, resp.Err
+			}
+			return resp.AuthorityIDs, nil
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		}
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	}
+}
+
+// Close closes the service
+func (s *Service) Close() error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if s.closed {
+		return nil
+	}
+	s.closed = true
+	// Note: Closing s.toWorker would typically be done by the owner of the channel
+	// or when the worker itself is shut down. The service just stops sending.
+	return nil
+}
```

Comment:
remove the return since the function does not return an error
```suggestion
func (s *Service) Close() {
	s.mu.Lock()
	defer s.mu.Unlock()
	
	// Note: Closing s.toWorker would typically be done by the owner of the channel
	// or when the worker itself is shut down. The service just stops sending.
	if s.closed {
		return
	}
	s.closed = true
}
```

Reply:
Accepted